addons/mail/static/src/models/composer/composer.js

```diff
@@ -19,18 +19,66 @@ function factory(dependencies) {
         // Public
         //----------------------------------------------------------------------
 
-        closeMentionSuggestions() {
+        closeSuggestions() {
+            if (this.activeElementName) {
+                this.update({
+                    [this.activeElementName]: [['unlink']],
+                });
+            }
+            if (this.extraListName) {
+                this.update({
+                    [this.extraListName]: [['unlink-all']],
+                });
+            }
+            if (this.mainListName) {
+                this.update({
+                    [this.mainListName]: [['unlink-all']],
+                });
+            }
             this.update({
-                activeSuggestedPartner: [['unlink']],
-                extraSuggestedPartners: [['unlink-all']],
-                mainSuggestedPartners: [['unlink-all']],
+                activeElementName: "",
+                delimiter: "",
+                extraListName: "",
+                mainListName: "",
             });
         }
 
         /**
          * Hides the composer, which only makes sense if the composer is
          * currently used as a Discuss Inbox reply composer.
          */
+        detectDelimiter() {
+            const delimiter = this.textInputContent.substring(this.textInputCursorStart - 1, this.textInputCursorStart);
+            const delimiters = ['@', ':', '#', '/'];
+            if(delimiters.includes(delimiter) && !this.hasSuggestions) {
+                this.update({ delimiter: delimiter });
+            }
+            const mentionKeyword = this._validateMentionKeyword(false);
+            if (mentionKeyword !== false) {
+                switch(this.delimiter) {
+                    case '@':
+                        this.update({
+                            activeElementName: "activeSuggestedPartner",
+                            elementModel: "mail.partner",
+                            extraListName: "extraSuggestedPartners",
+                            mainListName: "mainSuggestedPartners",
+                        });
+                        this._updateSuggestedPartners(mentionKeyword);
+                        break;
+                    case ':':
+                        this.update({
+                            activeElementName: "activeSuggestedCannedResponse",
+                            elementModel: "mail.canned_response",
+                            mainListName: "suggestedCannedResponses",
+                        });
```

Comment:
Shouldn't this case force write `extraListName` to empty in case it was set before?